builtin/nomad/task.go

```diff
@@ -0,0 +1,265 @@
+task {
+	use "nomad" {}
+}
+`)
+
+	doc.SetField(
+		"region",
+		"The Nomad region to deploy the on-demand runner task to.",
+		docs.Default(defaultODRRegion),
+	)
+
+	doc.SetField(
+		"datacenter",
+		"The Nomad datacenter to deploy the on-demand runner task to.",
+		docs.Default(defaultODRDatacenter),
+	)
+
+	doc.SetField(
+		"namespace",
+		"The Nomad namespace to deploy the on-demand runner task to.",
+		docs.Default(defaultODRNamespace),
+	)
+
+	doc.SetField(
+		"resources_cpu",
+		"Amount of CPU in MHz to allocate to this task. This can be overriden with "+
+			"the '-nomad-runner-cpu' flag on server install.",
+		docs.Default(fmt.Sprint(defaultODRCPU)),
+	)
+
+	doc.SetField(
+		"resources_memory",
+		"Amount of memory in MB to allocate to this task. This can be overriden with "+
+			"the '-nomad-runner-memory' flag on server install.",
+		docs.Default(fmt.Sprint(defaultODRMemory)),
+	)
+
+	doc.SetField(
+		"nomad_host",
+		"Hostname of the Nomad server to use for launching on-demand tasks.",
+		docs.Default(defaultNomadHost),
+	)
+
+	return doc, nil
+}
+
+// Config implements Configurable.
+func (p *TaskLauncher) Config() (interface{}, error) {
+	return &p.config, nil
+}
+
+// StopTask signals to Nomad to stop the nomad job created previously.
+func (p *TaskLauncher) StopTask(
+	ctx context.Context,
+	log hclog.Logger,
+	ti *TaskInfo,
+) error {
+	client, err := p.getNomadClient()
+	if err != nil {
+		log.Error("failed to create a Nomad API client to stop an ODR task")
+		return err
+	}
+
+	_, _, err = client.Jobs().Deregister(ti.Id, true, nil)
+	return err
+}
+
+// StartTask creates a Nomad job for working on the task.
+func (p *TaskLauncher) StartTask(
+	ctx context.Context,
+	log hclog.Logger,
+	tli *component.TaskLaunchInfo,
+) (*TaskInfo, error) {
+	client, err := p.getNomadClient()
+	if err != nil {
+		log.Error("failed to create a Nomad API client to start an ODR task")
+		return nil, err
+	}
+
+	// Generate an ID for our pod name.
+	id, err := ulid.New(ulid.Now(), rand.Reader)
+	if err != nil {
+		return nil, err
+	}
+
+	// Generate unique task name
+	taskName := strings.ToLower(fmt.Sprintf("waypoint-task-%s", id.String()))
+
+	// Set some defaults
+	if p.config.Region == "" {
+		p.config.Region = defaultODRRegion
+	}
+	if p.config.Datacenter == "" {
+		p.config.Datacenter = defaultODRDatacenter
+	}
+	if p.config.Namespace == "" {
+		p.config.Namespace = defaultODRNamespace
+	}
+	if p.config.Memory == 0 {
+		p.config.Memory = defaultODRMemory
+	}
+	if p.config.CPU == 0 {
+		p.config.CPU = defaultODRCPU
+	}
+	if p.config.NomadHost == "" {
+		p.config.NomadHost = defaultNomadHost
+	}
+
+	log.Trace("creating Nomad job for task")
+	jobclient := client.Jobs()
+	job := api.NewServiceJob(taskName, taskName, p.config.Region, 10)
+	job.Datacenters = []string{p.config.Datacenter}
+	tg := api.NewTaskGroup(taskName, 1)
+	tg.Networks = []*api.NetworkResource{
+		{
+			Mode: "host",
+		},
+	}
+
+	job.Namespace = &p.config.Namespace
```

Comment:
@briancain I think this line could be causing some unexpected behavior. In my runner profile, I set the Nomad namespace to be "ops" (where I keep my infra/devops-related Nomad jobs, such as a Waypoint runner). The Nomad jobspec that the runner is deploying is written to deploy to the "dev" namespace, since it's an application. However, the result that I'm seeing is that the Nomad job which is deployed ends up in the namespace of the runner, which is "ops".

Runner profile:
```
{
  "datacenter": "dc1",
  "namespace": "ops",
  "nomad_host": "http://nomad.service.dc1.consul:4646",
  "region": "global",
  "resources_cpu": "2000",
  "resources_memory": "4000"
}
```
Job snippet:
```
job "my-job" {
  name        = "my-job"
  namespace   = "${namespace}"  //the value being passed here via templatefile in waypoint.hcl is "dev", which is also the Waypoint workspace name
```